Add Template and Action classes and tests

mvriel opened the pull request.

jaapio commented on src/phpDocumentor/Renderer/ActionHandlerLocator.php:
```diff
@@ -0,0 +1,48 @@
+class ActionHandlerLocator
+{
+    /**
+     * @var ContainerInterface
```
missing a summary here

jaapio commented on tests/unit/phpDocumentor/Renderer/ActionHandlerLocatorTest.php:
```diff
@@ -0,0 +1,34 @@
+{
+    public function testIfLocatorRetrievesAnActionWithHanderAppendedToClassName()
+    {
+        $action = m::mock('phpDocumentor\Renderer\Action');
```
I would use the ::class constant here, Helps when you refactor the name of a class

jaapio commented on src/phpDocumentor/Renderer/Template.php:
```diff
@@ -0,0 +1,120 @@
+ * With a template you can provide a series of parameters that can be passed to every Action
+ * so that a default piece of information is provided, such as the target location.
+ */
+class Template
```
shouldn't this be final?

mvriel commented:
@jaapio thanks for the review; I have fixed your feedback and added the missing test for the Template class (had forgotten to commit that in the initial commit *facepalm*)

jaapio commented:
looks good, Wercker should be able to pass? think you need a rebase here. But can be merged without that.
